### DISCUSSION TITLE
New ruleset idea - osu!prism

### DISCUSSION THREAD

**Discussion by Tenexxt** [ppy/osu#28867] [Ideas]
**Osu!prism** (aka the prototype name for this mode) would have similar controls to catch, exept, as already said - things are hurting you instead of, well, being things to collect. While i don't play catch, i like how its made. In my opinion, its the hardest one of the sub modes in osu, since you don't click to the rythm, but you collect to the rythm.

In prism, **the gameplay looks like a triangle** and requries a high reading skill. (_Note: I got that from Lazers theme: 'triangles', it wasn't random + **the image on how it would look can be seen later in this post**_)

The triange would be a bit higher from the center, and the **player mainly controls the part facing downwards**. on the down part, you can control a similar thing that you do in catch, exept its a circle (so its more visible, and clear when you got hit and not confusing to the player), and the objects to avoid would go from down to up (the same way as in 'Floating fruits' mod in catch) There are two main things to avoid:
- **Zones** - They would work almost the same to this attack from Sans in Undertale: https://drive.google.com/file/d/1aieMcz8_DuqJ9I5Jd-7n9HVwFSKjeAcd/view?usp=sharing They are adjustable to be tight gaps like this, or just so the player has to play on a smaller field and have a harder time avoiding the second object:
-  **Prism** (also called 'Note' through ot the post, because its just a fancy name for it), which is a red triangle that 'falls' on you.
 _Note: After some thoughts, i decided to keep the dash from catch for this mode (its listed as a note, because i said the opposite earlier)_

**"Okay, but what about the whole triangle thing? what about the 2 other sides?"** Basically, on the sides there are:
- **Lazers** (at first i named them like that out of desperation, but the name stuck), which you click on time, kinda like in osu standard. **'Why its closer to standard rather to taiko or mania?' because they expand on the sides, and when they reach a certain size, you click,** nothing is falling on the sides that you click on, like in taiko or mania - but they kinda apear like o!s circles (more visable in the concept art)
 _Note: they aren't supposed to be the main focus of the gameplay, so it would be advised to not be spammed in harder sections, and just used as a bass/scratch sound (unless we are talking harder maps ofc)_

Also, on the down part, you can occur a rotater , which rotates the triangle in a way, so you play basically on the other side, but no in betweens (so the player doesn't have a different rotation for every map, also this can be erased). its a undogable rectangle, so its not something that can be avoided. It has a purple colour, and if you wonder how would it look in-game to ecounter it, you would see a purple rectangle coming to the center, but **'What about the objects that appear soon later'** - In case the player missed the rotater, they will most likely understand whats going on when they see prisms and zones on the other side of the triangle. 
_Note: Rotaters **always spin** the triangle counter-clockwise_

**Overall, the gameplay would look somewhere like this; https://drive.google.com/file/d/1jIMI2fsRaUvpCjWi5xJr-s0YvAlJDDSP/view?usp=sharing**

The main colour of hit objects is red, but in my opinion is good to add a veriety of colours, so Prisms and notes are distinguishable - thats when combo colours come in, Prisms can have seperate colour combo group, so that the colours are distinguishable

_Note: in the concept lazers are looking as cut triangles, but thats because i was making the image fast and wanted atleast some level of symmetry, they are supposed to be rectangles + In my opinion it would be good to make the main triangle bigger, but its a concept and it can be changed after all. _

**What about hitsounds?**
- The lazers and notes are hitsoundable. if you dont click on the lazer, no hs. if you get hit by a note, no hs. if the note is dodged, and hits the bar (as intended) and not the ball (you), there is a hs, that the note is supposed to make

**How about the icon?**
- Something like this would be fine: https://drive.google.com/file/d/1JgkY4TL9MDIdMeTr47NcipvP3zHe95dg/view?usp=sharing
_Note: view this on a dark background, or invert it: Its drawn in white_

**What about stats?**

- Approach rate: Speed the notes and zones aproach in
- HP drain: Do i really have to explain?
- Accuracy/OD: Changes the speed the zones damage you in, changes note size, and the timing range for lazers
- Lazer range: Like said before, **Lazers are supposed to be hit at a certain size, not when they reach the corrner of the triangle**. This can be seen in the concept image of the mode, but it can be adjusted to be a smaller range, or a bigger one) 1- biggest range 10 - smallest range

**Scoring system:**

**Lazers**: you can get a: great, good, meh, miss (like osu standard) click on the beat basically. it has an accuracy score like in other modes
_Notes and zones_: they have their own accuracy % and you can get a Great and miss, Great if dodged a note and for every beat outside a zone (if there is a one, thats also why zones have to last a whole beat atleast), miss if got hit by a note/ entered a zone. you also get a miss every x in a zone ( x, as in unknown number that is changed my the OD/Accuracy stat of a map

**In-depth on the basic controls**

O or Z- Lazer 1
P or X- Lazer 2
LShift or C - Dash
Q or Left Arrow - Move left
W or Right Arrow - Move right

(can be changed)

I have realised that I don't know how to code at all, so I can't make this a custom ruleset, but if anyone wants to make this - you are allowed to do so, just contact me first so I'll know about it and could give some pointers (mostly to design)

I understand if this wont be considered, because programing hell, and making new medals, and the list goes on, but either way:
Thanks for reading about the concept of **osu!prism**

Edit: Added some improvements to the posts format, and deleted some useless features
Edit2: Explained objects more and added some overall QoL features to o!p, also some more formating to the post was made
Edit3: Changed the modes name because it's already taken (sorry if I missed the name somewhere in the post and the old one is still there)

**Comment by shiumano** [comment]
Sorry to complain about the temporary name, but a rule set with this name already exists (Muse Dash clone).
https://github.com/Beamographic/rush
However, it is so confusing that we would like to come up with another name.

Since we have a good name, how about osu!prism?

**Reply by Tenexxt** [reply]
Yea, it sounds like a better name, It's definitely more fitting. Id keep looking for a better name still, but this one is still good in my opinion .
Already made changes to the main post, sorry if I missed some other mentions of it, it's 3 am T-T
